packages/react-core/src/components/DescriptionList/DescriptionList.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/DescriptionList/description-list';
+
+export interface DescriptionListProps extends Omit<React.HTMLProps<HTMLDListElement>, 'type'> {
+  /** Anything that can be rendered inside of the list */
+  children?: React.ReactNode;
+  /** Additional classes added to the list */
+  className?: string;
+  /** Sets the description list component term and description pair to a horizontal layout. */
+  isHorizontal?: boolean;
+  /** Sets the description list to format automatically. */
+  hasAutoColumnWidths?: boolean;
+  /** Modifies the description list display to inline-grid. */
+  hasInlineGrid?: boolean;
+  /** Sets the number of columns on the description list */
+  variant?: '1Col' | '2Col' | '3Col';
```

Comment:
Since the columns are breakpoint dependent,  would use the interface we use in other components.  Look at Tabs `inset` prop  for an example. I also would rename it.  It is not really a variant.